model/task/db.go

```diff
@@ -203,12 +229,7 @@ var (
 						"$and": []bson.M{
 							{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
 							{"$eq": []interface{}{"$" + ActivatedKey, true}},
-							{
-								"$or": []bson.M{
-									{DependsOnKey: 0},
-									{"$ne": []interface{}{"$" + bsonutil.GetDottedKeyName(DependsOnKey, DependencyUnattainableKey), true}},
-								},
-							},
+							{"$ne": []interface{}{"$" + HasUnattainableDepsKey, true}},
```

Comment:
Is this the only place we use HasUnattainableDeps? In which case we don't need to add it to the task document, since everything is done within the aggregation. You can just use a local global variable string or something.

Reply:
Good call